cli/towerSysUtil.ml

```diff
@@ -103,5 +105,41 @@ module Dir = struct
       match err_opt with None -> make_exist ~perm path | Some err -> Some err
     ) None
 
+  (* Placeholder for now *)
+  let log msg = ()
+
+  let gather_nodes root =
+    (* By default enforce all recursive processing is done relative to
+    the cwd dir, unless overridden *)
+    let cwd = Unix.getcwd() in
+    let rec gather_nodes node =
+      let node_path = cwd ^ "/" ^ node in
+      match (Unix.lstat node_path).st_kind with
+      | Unix.S_REG -> [|node|]
+      | Unix.S_DIR -> let subnodes = (Sys.readdir node) in
+                      let descend subnode =
+                        let subnode_path = node ^ "/" ^ subnode in
+                        gather_nodes subnode_path
+                      in
+                      Array.fold_left Array.append [|node|] (Array.map descend subnodes)
+      | _ -> [||]
```

Comment:
I think this means that symlinks will never be listed and removed (and then `rmdir` will fail).